### PR TITLE
Livesync alpha docs

### PR DESCRIPTION
**Pull request by franrob-projects**
This PR:

- Adds LiveSync as a new product type
- Syntax highlighting for [SQL]
- Adds content for the LiveSync product

### CODE REVIEW

File: content/livesync/connector/index.textile [hunk 1]
```diff
@@ -0,0 +1,336 @@
+---
+title: Deploy
```
Comment (mschristensen) [inline content/livesync/connector/index.textile:2]: I don't think this is the right terminology for this page. The "Deploy to Railway" heading is fine, as that is an actual example deployment to Railway cloud. However the rest of it is about running locally with docker compose (or just pulling the container image), not deployment.

File: content/livesync/index.textile [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+- Database Connector := A backend service for broadcasting changes from your database in realtime.
+- Models SDK := A frontend library that simplifies subscribing to changes in data models and applying optimistic updates, merging them with confirmed updates.
+
+The following diagram provides a simplified overview of the LiveSync solution:
+
+h2(#db-connector). Database Connector
+
+The "Database Connector":/livesync/connector uses the "outbox pattern":https://docs.aws.amazon.com/prescriptive-guidance/latest/cloud-design-patterns/transactional-outbox.html to broadcast application-specific change events from your database to frontend clients via "Ably Channels":/channels.
+
+When you update the data in your database, you record a change event transactionally in a dedicated "outbox table":https://docs.google.com/document/d/19BJ4oYC3ah4wFH0VS5Qv6qHe3qmRS_PpNLSytTOi9tc/edit#heading=h.r00ywdkts746. The Database Connector detects new changes in the outbox and publishes them as messages on Ably Channels. Using the Models SDK, your client applications subscribe to these change events and update their local state with the new data.
```
Comment (mschristensen) [inline content/livesync/index.textile:20]: This shouldn't link to the google doc, but rather to the outbox table section on the docs site

File: content/livesync/connector/index.textile [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+meta_description: "The Ably Database Connector is available as a Docker container. Deploy it locally using Docker Compose, or on any cloud infrastructure that supports Docker images."
+product: livesync
+---
+
+The Database Connector uses the "outbox pattern":https://docs.aws.amazon.com/prescriptive-guidance/latest/cloud-design-patterns/transactional-outbox.html to broadcast application-specific change events from your database to frontend clients via "Ably Channels":/channels.
+
+When you update the data in your database, you record a change event transactionally in a dedicated "outbox table":/livesync/connector/database#outbox. The Database Connector detects new changes in the outbox and publishes them as messages on Ably Channels. Using the Models SDK, your client applications subscribe to these change events and update their local state with the new data.
+
+The Database Connector is available as a "Docker":/https://www.docker.com/ container image that can be deployed on any cloud infrastructure platform that supports running Docker container images.
```
Comment (mschristensen) [inline content/livesync/connector/index.textile:11]: I don't think we want a leading slash on the link here, as it resolves to an incorrect relative path on the docs site: https://ably-docs-livesync-alph-pcjpdk.herokuapp.com/docs/https://www.docker.com

File: content/livesync/connector/index.textile [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+```
+
+h3(#docker-configure). Configure in Docker Compose
+
+There are three "options":#options to configure the Database Connector. The following examples will use environment variables.
+
+Create a file called @adbc.env@ with at least the following configuration options:
+
+```[sh]
+ADBC_ABLY_API_KEY={{API_KEY}}
```
Comment (mschristensen) [inline content/livesync/connector/index.textile:81]: This is interpolating an actual API key when I preview the docs site... is this expected?

image

File: content/livesync/connector/index.textile [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+Open your project in the Railway console:
+
+```[sh]
+railway open
+```
+
+Select the “adbc-test” service, then navigate to *Variables* -> *RAW Editor*. Paste the following variables into this section:
+
+```[sh]
+ADBC_ABLY_API_KEY={{API_KEY}}
```
Comment (mschristensen) [inline content/livesync/connector/index.textile:191]: Same here

File: content/livesync/connector/index.textile [hunk 6]
```diff
@@ -0,0 +1,336 @@
...
+<aside data-type='note'>
+<p>At least an Ably API key and the connection details of your database must be supplied to correctly configure the connector.</p>
+</aside>
+
+The following table provides descriptions for the most commonly used configuration options:
+
+|_. Option |_. Description |
+| @ADBC_ABLY_API_KEY@ | your Ably "API key":/auth#api-keys. |
+| @ADBC_POSTGRES_CONNECTION_URI@ | the full connection URI of your Postgres database. |
+| @ADBC_POSTGRES_HOST@ | your Postgres database host name as an alternative to providing the @CONNECTION_URI@. |
```
Comment (mschristensen) [inline content/livesync/connector/index.textile:246]: These should be `ADBC_POSTGRES_CONNECTION_URI` instead of `CONNECTION_URI`

File: content/livesync/connector/index.textile [hunk 7]
```diff
@@ -0,0 +1,336 @@
...
+| @ADBC_POSTGRES_PORT@ | your Postgres database port number as an alternative to providing the @CONNECTION_URI@. |
+| @ADBC_POSTGRES_DATABASE@ | your Postgres database name as an alternative to providing the @CONNECTION_URI@. |
+| @ADBC_POSTGRES_USER@ | your Postgres database user as an alternative to providing the @CONNECTION_URI@. |
+| @ADBC_POSTGRES_PASSWORD@ | your Postgres database user password as an alternative to providing the @CONNECTION_URI@. |
+| @--config@ | can only be specified as a CLI flag and enables you to override the path to a YAML configuration file. |
+| @ADBC_ENV@ | is an environment descriptor (either development or production). "development" pretty-prints log output with logging enabled at debug level and above. "production" logs output in JSON format with logging enabled at info level and above. |
+| @ADBC_LOG_LEVEL@ | specifies the log level to use (one of: debug, info, warn, error, fatal, panic) and overrides any presets from ADBC_ENV. |
```
Comment (mschristensen) [inline content/livesync/connector/index.textile:253]: `ADBC_ENV` should be code formatted with `@ADBC_ENV@`

File: content/livesync/connector/index.textile (hunk 5 above)
Comment (mschristensen) [reply content/livesync/connector/index.textile:191]: Also elsewhere, but I'll stop pointing this out assuming it's intended :)

File: content/livesync/connector/database.textile [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+CREATE ROLE YOUR_USER LOGIN PASSWORD 'your_database' VALID UNTIL 'infinity';
+GRANT CONNECT ON DATABASE your_database to your_user_USER;
+GRANT SELECT ON your_outbox_schema.your_outbox_table to your_user;
+GRANT DELETE ON your_outbox_schema.your_outbox_table to your_user;
+GRANT SELECT ON your_nodes_schema.your_nodes_table to your_user;
+GRANT DELETE ON your_nodes_schema.your_nodes_table to your_user;
+```
+
+<aside data-type='note'>
+<p>Additional privileges may be required when using other utility commands in the Database Connector, such as @CREATE TABLES@.</p>
```
Comment (mschristensen) [inline content/livesync/connector/database.textile:104]: This refers to the CLI command `create tables` which should be lowercase

Comment (mschristensen) [reply content/livesync/connector/database.textile:104]: We should also link to the relevant heading

File: content/livesync/connector/database.textile [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+* You can distribute change notifications across various "channels":/channels according to the requirements of your application.
+* Channels provide a mechanism for data access control via channel "capabilities":/auth/capabilities.
+
+h3(#schema). Schema
+
+The following table describes the schema of the outbox table in your database:
+
+|_. Field |_. Type |_. Required on write? |_. Example Value |_. Description |
+| @sequence_id@ | Serial integer (primary key) | No (automatically assigned) | @1@ | Monotonically increasing identifier that determines publish order within the scope of the channel. |
+| @mutation_id@ | String | Yes | @680f3f78-21ec-49a0-be99-25f89a84f232@ | TThe ID of the mutation, used for correlating the outbox event with an optimistic event applied locally on the client. |
```
Comment (mschristensen) [inline content/livesync/connector/database.textile:157]: Typo on `TThe`

File: content/livesync/connector/database.textile [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+| @channel@ | String | Yes | @documents@ | The "channel:/channels name on which to publish this change event. |
```
Comment (mschristensen) [inline content/livesync/connector/database.textile:158]: `"channel:/channels` link is broken (missing close quote?)

File: content/livesync/models/index.textile [hunk 11]
```diff
@@ -0,0 +1,203 @@
...
+
+<aside data-type='important'>
+<p>The examples use "basic authentication":/auth/basic to demonstrate features for convenience. In your own applications, basic authentication should never be used on the client-side as it exposes your Ably API key. Instead use "token authentication":/auth/token. </p>
+</aside>
+
+"Sign up":https://ably.com/sign-up to Ably to create an API key in the "dashboard":https://ably.com/dashboard or use the "Control API":/account/control-api to create an API programmatically.
+
+API keys and tokens have a set of capabilities assigned to them that specify which operations, such as @subscribe@ or @publish@ can be performed on which resources. To use the Models SDK, the API key requires the following "capabilities":/auth/capabilities
+
+* @Subscribe@ for the channels you intend to subscribe to.
```
Comment (mschristensen) [inline content/livesync/models/index.textile:23]: `subscribe` should be lowercase as it refers to a specific capability string. Same for `history` below

File: content/livesync/models/index.textile [hunk 12]
```diff
@@ -0,0 +1,203 @@
...
+```[javascript]
+const client = new Realtime.Promise({ key: '{{API_KEY}}' });
+const modelsClient = new ModelsClient({ client });
+```
+
+h3(#options). ClientOptions
+
+In addition to the underlying Ably realtime client, you can provide a number of other "@ClientOptions@":/api/realtime-sdk/types#client-options to configure the behavior of the Models SDK:
+
+- @syncOptions@ := is used to configure how the model state is synchronised via the Sync function.
```
Comment (mschristensen) [inline content/livesync/models/index.textile:54]: should link to sync function heading

File: content/livesync/models/index.textile [hunk 13]
```diff
@@ -0,0 +1,203 @@
...
+
+The state of our model will have the following type:
+
+```[javascript]
+type Post = {
+  id: number;
+  text: string;
+  comments: string[];
+};
+```
```
Comment (mschristensen) [inline content/livesync/models/index.textile:95]: These lines should appear under "Create a sync function" like:

> Create a simple sync function which loads the post and its comments.
> The state of our model will...
> The Models SDK will infer the type of the model state from the type of the data payload returned by the sync function.

File: content/livesync/models/models.textile [hunk 14]
```diff
@@ -0,0 +1,291 @@
...
+COMMIT;
+```
+
+For this to work correctly, use a "Read Committed":https://www.postgresql.org/docs/7.2/xact-read-committed.html transaction isolation level when reading your database state. This is the default isolation level in PostgreSQL.
+
+h3(#history). Replay from history
+
+After the Models SDK initializes the model state via the sync function, it is important to apply the exact sequence of change events from your backend from the correct point in the stream. This position is specified by the @sequenceID@ returned from your backend endpoint.
+
+Internally, after calling the sync function, the SDK attaches to the "channel":/channels and "subscribes":#subscribe to live messages. Live messages are buffered and not yet processed. It then paginates backwards through the message history on the channel from the point of attachment using "@untilAttach@":/storage-history/history#until-attach. The number of items in each page can be configured via the @syncOptions.historyPageSize@ defined in your "@ClientOptions@":/livesync/models#options.
```
Comment (mschristensen) [inline content/livesync/models/models.textile:98]: The `subscribe` link here should be to _channel_ subscribe docs, not model subscribe docs

File: content/livesync/models/models.textile [hunk 15]
```diff
@@ -0,0 +1,291 @@
...
+}
+```
+
+Be aware that updating the model state in your database happens entirely via your backend endpoint.
+
+h3(#mutation). MutationID
+
+The Models SDK needs a way to know when a change event received from the backend corresponds to an optimistic event that has been applied locally.
+
+To do this, your clients must define a unique @mutationID@ on the optimistic event. This ID can be any string, but typically this will be a UUID generated from a library, such as "uuid"https://www.npmjs.com/package/uuid. The @mutationID@ should then be passed to your backend and included on the confirmation event it writes to the "outbox":/livesync/connector/database#outbox, for example:
```
Comment (mschristensen) [inline content/livesync/models/models.textile:204]: The link on the `uuid` is broken, I think missing a colon

**Review by mschristensen — CHANGES_REQUESTED** [review]
This is looking really great, did a last review and just some small comments left :) thanks!

File: content/livesync/connector/index.textile (hunk 1 above)
Comment (franrob-projects) [reply content/livesync/connector/index.textile:2]: Run the Connector

Comment (franrob-projects) [reply content/livesync/connector/index.textile:2]: or install @mschristensen

File: content/livesync/models/index.textile (hunk 13 above)
Comment (franrob-projects) [reply content/livesync/models/index.textile:95]: I'm not sure what you mean here sorry @mschristensen

Comment (mschristensen) [reply content/livesync/models/index.textile:95]: The type of the model is not relevant until we talk about the sync function, as the type of the model is inferred from the value of the `data` field returned by the sync function. So, I'm just suggesting that we move these lines L87-95 to appear under the "Create a sync function" heading on L115. My recommendation is that the new text under the "Create a sync function" heading is as described in my comment above. :)

File: content/livesync/index.textile (hunk 2 above)
Comment (franrob-projects) [reply content/livesync/index.textile:20]: Fixed in this [commit](https://github.com/ably/docs/pull/2082/commits/15a2be26866049471006c24c9e1e9094c5b53a85)

File: content/livesync/connector/index.textile (hunk 1 above)
Comment (franrob-projects) [reply content/livesync/connector/index.textile]: Fixed in this [commit](https://github.com/ably/docs/pull/2082/commits/15a2be26866049471006c24c9e1e9094c5b53a85)

File: content/livesync/connector/index.textile (hunk 3 above)
Comment (franrob-projects) [reply content/livesync/connector/index.textile]: Fixed in this [commit](https://github.com/ably/docs/pull/2082/commits/15a2be26866049471006c24c9e1e9094c5b53a85)

File: content/livesync/connector/index.textile (hunk 7 above)
Comment (franrob-projects) [reply content/livesync/connector/index.textile]: Fixed in this [commit](https://github.com/ably/docs/pull/2082/commits/15a2be26866049471006c24c9e1e9094c5b53a85)

File: content/livesync/connector/database.textile (hunk 9 above)
Comment (franrob-projects) [reply content/livesync/connector/database.textile]: Fixed in this [commit](https://github.com/ably/docs/pull/2082/commits/15a2be26866049471006c24c9e1e9094c5b53a85)

File: content/livesync/models/index.textile (hunk 11 above)
Comment (franrob-projects) [reply content/livesync/models/index.textile]: Fixed in this [commit](https://github.com/ably/docs/pull/2082/commits/c112ade0468d905e588f441cff4a895bd9b30ace)

File: content/livesync/models/index.textile (hunk 12 above)
Comment (franrob-projects) [reply content/livesync/models/index.textile]: https://github.com/ably/docs/pull/2082/commits/c112ade0468d905e588f441cff4a895bd9b30ace

File: content/livesync/models/index.textile (hunk 13 above)
Comment (franrob-projects) [reply content/livesync/models/index.textile]: Fixed in this [commit](https://github.com/ably/docs/pull/2082/commits/c112ade0468d905e588f441cff4a895bd9b30ace)

File: content/livesync/models/models.textile (hunk 14 above)
Comment (franrob-projects) [reply content/livesync/models/models.textile]: Fixed in this [commit](https://github.com/ably/docs/pull/2082/commits/c112ade0468d905e588f441cff4a895bd9b30ace)

File: content/livesync/models/models.textile (hunk 15 above)
Comment (franrob-projects) [reply content/livesync/models/models.textile]: Fixed in this [commit](https://github.com/ably/docs/pull/2082/commits/c112ade0468d905e588f441cff4a895bd9b30ace)

**Review by mschristensen — APPROVED** [review]

File: content/livesync/models/index.textile [hunk 16]
```diff
@@ -122,8 +112,17 @@ async function sync(id: number, page: number) {
...
 
-The following is an example response from the sync function:
+The model's relevance is determined by inspecting the data field returned by the sync function:
```
[the remark this answers is not in the document]
Comment (franrob-projects) [reply content/livesync/models/index.textile]: Fixed here https://github.com/ably/docs/pull/2082/commits/d09bc12cd6f286ffb7ab1232b1fa3d5ac7677e23  @mschristensen

File: content/livesync/models/models.textile [hunk 17]
```diff
@@ -0,0 +1,295 @@
...
+languages:
+  - javascript
+---
+
+A model is a single instance of a live, observable data model backed by your database. A model requires a "sync function":#sync to tell the model how to initialize with the latest data, and a "merge function":#merge to calculate the next version of the model state when change events are received from the backend database.
+
+You can then "subscribe":#subscribe to a model to receive updates in realtime whenever the model state changes.
+
+<aside data-type='note'>
+<p>You can access the Models SDK directly by referring to ABly's "API references":https://sdk.ably.com/builds/ably-labs/models/main/typedoc/.</p>
```
Comment (m-hulbert) [inline content/livesync/models/models.textile:14]: Capital `AB` on Ably.

I'd also suggest rewording this to something more simple, e.g: "View the API reference for the Models SDK".

**Review by m-hulbert — CHANGES_REQUESTED** [review]
1 minor comment 🙂

File: content/livesync/models/models.textile (hunk 17 above)
Comment (franrob-projects) [reply content/livesync/models/models.textile]: done!

Comment (franrob-projects) [reply content/livesync/models/models.textile]: 2ed8a89002398d33e43e2be6b052818b7237967d

**Review by m-hulbert — APPROVED** [review]
Nice work! 🎉
